internal/pkg/service/appproxy/appconfig/appconfig_test.go

```diff
@@ -0,0 +1,319 @@
+					},
+					expectedConfig: AppProxyConfig{
+						ID:              "6",
+						Name:            "6",
+						UpstreamAppHost: "app.local",
+						eTag:            `"etag-value"`,
+						maxAge:          60 * time.Second,
+					},
+				},
+				{
+					delay: 10 * time.Minute,
+					responses: []*http.Response{
+						newResponse(t, 200, map[string]any{"id": "6", "upstreamAppHost": "new-app.local"}, `"etag-new-value"`, "max-age=60"),
+					},
+					expectedETag: `"etag-value"`,
+					expectedConfig: AppProxyConfig{
+						ID:              "6",
+						Name:            "6",
+						UpstreamAppHost: "new-app.local",
+						eTag:            `"etag-new-value"`,
+						maxAge:          60 * time.Second,
+					},
+				},
+				{
+					expectedConfig: AppProxyConfig{
+						ID:              "6",
+						Name:            "6",
+						UpstreamAppHost: "new-app.local",
+						eTag:            `"etag-new-value"`,
+						maxAge:          60 * time.Second,
+					},
+				},
+			},
+		},
+		{
+			name:  "etag-error",
+			appID: "7",
+			attempts: []attempt{
+				{
+					responses: []*http.Response{
+						newResponse(t, 200, map[string]any{"id": "7", "upstreamAppHost": "app.local"}, `"etag-value"`, "max-age=60"),
+					},
+					expectedConfig: AppProxyConfig{
+						ID:              "7",
+						Name:            "7",
+						UpstreamAppHost: "app.local",
+						eTag:            `"etag-value"`,
+						maxAge:          60 * time.Second,
+					},
+				},
+				{
+					delay: 10 * time.Minute,
+					responses: []*http.Response{
+						newResponse(t, 500, map[string]any{}, "", ""),
+						newResponse(t, 500, map[string]any{}, "", ""),
+						newResponse(t, 500, map[string]any{}, "", ""),
+						newResponse(t, 500, map[string]any{}, "", ""),
+						newResponse(t, 500, map[string]any{}, "", ""),
+						newResponse(t, 500, map[string]any{}, "", ""),
+					},
+					expectedETag: `"etag-value"`,
+					expectedConfig: AppProxyConfig{
+						ID:              "7",
+						Name:            "7",
+						UpstreamAppHost: "app.local",
+						eTag:            `"etag-value"`,
+						maxAge:          60 * time.Second,
+					},
+				},
+				{
+					delay: time.Hour,
+					responses: []*http.Response{
+						newResponse(t, 500, map[string]any{}, "", ""),
+						newResponse(t, 500, map[string]any{}, "", ""),
+						newResponse(t, 500, map[string]any{}, "", ""),
+						newResponse(t, 500, map[string]any{}, "", ""),
+						newResponse(t, 500, map[string]any{}, "", ""),
+						newResponse(t, 500, map[string]any{}, "", ""),
+					},
+					expectedETag:      `"etag-value"`,
+					expectedErrorCode: 500,
+				},
+			},
+		},
+	}
+
+	t.Parallel()
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			clk := clock.NewMock()
+			transport := httpmock.NewMockTransport()
+
+			url := "https://sandboxes.keboola.com"
+			loader := NewLoader(log.NewDebugLogger(), clk, url)
+			loader.sender = loader.sender.(client.Client).WithTransport(transport)
```

Comment:
Please, inject the client as a dep. via constructor.

What if you need to mock the transport at a higher level?

You can also switch the test to blackbox mode, `package ..._test`, you will only test the public interface of the Loader, and it will lead to a better design.